tests/test_models.py

```diff
@@ -5,339 +5,295 @@
 """
 
 from contextlib import nullcontext as does_not_raise
-from logging import CRITICAL, ERROR, INFO, WARNING
+from logging import DEBUG, ERROR, INFO, WARNING
+from typing import Any
 
 import pytest
 from numpy import datetime64, timedelta64
 
-from virtual_rainforest.core.model import BaseModel, InitialisationError
-from virtual_rainforest.models.abiotic.model import AbioticModel
-from virtual_rainforest.models.soil.model import SoilModel
-
 from .conftest import log_check
 
 
-def test_base_model_initialization(caplog, mocker):
-    """Test `BaseModel` initialization."""
-
-    # Patch abstract methods so that BaseModel can be instantiated for testing
-    mocker.patch.object(BaseModel, "__abstractmethods__", new_callable=set)
-
-    # Initialise model
-    model = BaseModel(timedelta64(1, "W"), datetime64("2022-11-01"))
+@pytest.fixture(scope="module")
+def data_instance():
+    """Creates a simple data instance for use in testing."""
+    from xarray import DataArray
 
-    # In cases where it passes then checks that the object has the right properties
```

Comment:
Is this section being removed because the functionality is tested when inheriting classes are tested for this?

Reply:
Yup - I'm running this testing on an example minimal real class rather than making it possible to create an instance of BaseModel itself. The `test_check_failure_on_missing_methods` tackles this specifically. It would also catch undefined `model_name` and `required_init_vars` but those are checked in more detail in `__init_subclass__` so cannot  be missing when an instance is created.